### PR TITLE
Add index field pub_date_start_itsi

### PR DESCRIPTION
**Pull request by hackmastera**
advances #4254

### CODE REVIEW

File: app/indexers/facet_indexer.rb [hunk 1]
```diff
@@ -48,6 +49,14 @@ def structure?
     false
   end
 
+  def pub_date_start
+    date = resource.primary_imported_metadata.created
+    return unless date
```
Comment (tpendragon) [inline app/indexers/facet_indexer.rb:54]: Should this be `date.present?`? I assume created is a Set and could be an empty array.

File: app/indexers/facet_indexer.rb [hunk 2]
```diff
@@ -48,6 +49,14 @@ def structure?
...
+    date = date.first
+    date = Time.zone.parse(date) unless date.respond_to?(:year)
```
Comment (tpendragon) [inline app/indexers/facet_indexer.rb:56]: Should we capture parsing errors here or do you think we should just wait for it to happen? Could end up failing to save objects.

File: spec/indexers/facet_indexer_spec.rb [hunk 3]
```diff
@@ -11,8 +11,20 @@
 
         expect(output[:display_subject_ssim]).to eq scanned_resource.imported_metadata.first.subject
         expect(output[:display_language_ssim]).to eq ["English"]
+        expect(output[:pub_date_start_itsi]).to eq 1982
+      end
+
+      it "parses the first year from a date range" do
```
Comment (tpendragon) [inline spec/indexers/facet_indexer_spec.rb:17]: Do we need a similar test for a bibid date?

**Review by tpendragon — CHANGES_REQUESTED** [review]

File: app/indexers/facet_indexer.rb (hunk 1 above)
Comment (hackmastera) [reply app/indexers/facet_indexer.rb:54]: I added a test where the source doesn't have the created date and it goes okay. but `present?` works too and is more flexible so 👍

**Comment by hackmastera** [comment]
@tpendragon I added a bunch more error handling. should be ready for another look.

**Review by tpendragon — APPROVED** [review]

**Pull request merged**